docs/DeveloperGuide.md

```diff
@@ -1,30 +1,116 @@
 # Developer Guide
 
+---
+Welcome to the PlanPal Developer Guide! Thank you for taking an interest in the behind-the-scenes work of our product. We hope this document proves informative and useful for your work!
+
+## Table of Contents
+{to be inserted at a later date}
+
+---
+
 ## Acknowledgements
 
+
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Setting up, getting started
+1. Java 17 is required for PlanPal to function properly, please make sure your device has Java 17 installed. If not, you may download it [here](). If you need further assistance installing Java 17, you may refer to the [Installation Guide.]()
+2. Once you have ascertained that your device has Java 17 installed, you may download the `.jar file` [here]().
+insert SS of release page
+
+Click on `PlanPal.jar` to automatically download the file. Place the file in a folder your choice.
+
+
+
+## Design
+
+---
+### Architecture
+![Architecture.drawio.png](Images%2FArchitecture.drawio.png)
+
+The **Architecture Diagram** given above explains the high-level design of the program. Given below is a quick overview of the main components.
+
+### Main Components
```

Comment:
You can consider doing other UML diagrams for these components either Class / Object diagrams